Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Culling.h

```diff
@@ -68,6 +68,10 @@ namespace AZ
                 //! Set to all 0's if you don't want to hide the object from any Views.
                 RPI::View::UsageFlags m_hideFlags = RPI::View::UsageNone;
 
+                //! UUID and type of the component that owns this cullable (optional)
+                AZ::Uuid m_componentUuid = AZ::Uuid::CreateNull();
+                uint32_t m_componentType = 0;
```

Comment:
Can't this use the new Culling::ComponentType enum instead of uint32_t?

Reply:
This in the RPI level, whereas the ComponentType is in the Features level.  I used uint32_t here to avoid pushing features-specific component types into the RPI.

Reply:
How will we prevent component type IDs from colliding when multiple teams are making feature gems? Maybe this should be a Uuid too, and generate a random one for the reflection probes? Or use a registration system (maybe that's over-engineered)?

Reply:
Any component types that want to use the visualization system for queries will need to be added to RenderCommon.h at this time.